lib/conjur/base.rb

```diff
@@ -169,6 +169,13 @@ def initialize username, api_key, token, remote_ip = nil
       @username = username
       @api_key = api_key
       @token = token
+      
+      # Assume that the token was recently minted, log a warning if not
+      @token_born = gettime   
+      if token && (delta = Time.now - Time.parse(token['timestamp'])) > 10
+        $stderr.puts "WARNING: token timestamp offset by #{delta.to_i} seconds"
+      end
+
```

Comment:
Good to know, thanks.

On Wed, Jun 22, 2016 at 1:11 PM, Rafał Rzepecki <notifications@github.com>
wrote:

> In lib/conjur/base.rb
> <https://github.com/conjurinc/api-ruby/pull/77#discussion_r68094262>:
>
> > @@ -169,6 +169,13 @@ def initialize username, api_key, token, remote_ip = nil
> >        @username = username
> >        @api_key = api_key
> >        @token = token
> > +
> > +      # Assume that the token was recently minted, log a warning if not
> > +      @token_born = gettime
> > +      if token && (delta = Time.now - Time.parse(token['timestamp'])) > 10
> > +        $stderr.puts "WARNING: token timestamp offset by #{delta.to_i} seconds"
> > +      end
> > +
>
> A monotonic clock is more robust for measuring the passage of time. RTC
> isn't really appropriate for that: due to time synchronization and leap
> seconds it can jump back and forth, slow down or speed up. 22 cze 2016
> 13:02 "apotterri" <notifications@github.com> napisał(a):
> … <#m_-6880644014650230766_>
> In lib/conjur/base.rb <#77 (comment)
> <https://github.com/conjurinc/api-ruby/pull/77#discussion_r68033921>>: >
> @@ -169,6 +169,13 @@ def initialize username, api_key, token, remote_ip =
> nil > @username = username > @api_key = api_key > @token
> <https://github.com/token> = token > + > + # Assume that the token was
> recently minted, log a warning if not > + @token_born = gettime > + if
> token && (delta = Time.now - Time.parse(token['timestamp'])) > 10 > +
> $stderr.puts "WARNING: token timestamp offset by #{delta.to_i} seconds" > +
> end > + Seems like a simpler alternative might be to always use Time.now.
> What was the reason for using CLOCK_MONOTONIC? — You are receiving this
> because you were mentioned. Reply to this email directly, view it on GitHub
> <
> https://github.com/conjurinc/api-ruby/pull/77/files/6afb0002a561235aebb23f51414863c60eeccd98#r68033921>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/AAyRVM-U5hMbYaI_0rQGMrlWilGMjPoAks5qORYtgaJpZM4I7PQS>
> .
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/conjurinc/api-ruby/pull/77/files/6afb0002a561235aebb23f51414863c60eeccd98#r68094262>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEBHA5sSkCcyxqfz3DQYC4SgLddA0v-mks5qOWzdgaJpZM4I7PQS>
> .
>



-- 
*Alan Potter*
Engineer
p: 617.906.8240


<http://t.yesware.com/tt/58d6ef3f2477232db08043fa9198ec0f5a0fde51/f59ded25d9f0fae3e973353a29205cf4/5ff664c66c5990e35bd8d1f3bfc0b483/www.conjur.net/>

*Identify, Authorize, Audit... Everything*
Demo
<http://t.yesware.com/tt/58d6ef3f2477232db08043fa9198ec0f5a0fde51/f59ded25d9f0fae3e973353a29205cf4/13deaa1654e874c3dccc0acce4e70193/t.sidekickopen03.com/e1t/c/5/f18dQhb0S1KL8bGch0W2n0x6l2B9nMJW7t6jwn3LqWkjN2BgL6d1vVSKf69FLZ203?t=https%3A%2F%2Finfo.conjur.net%2Fdemo&si=3680029&pi=47d6aafc-d455-4a13-9ffc-a8737ef09b32>
 | Resources
<http://t.yesware.com/tt/58d6ef3f2477232db08043fa9198ec0f5a0fde51/f59ded25d9f0fae3e973353a29205cf4/cc9a5a824c8143be8a0bab0dcc7f1c20/t.sidekickopen03.com/e1t/c/5/f18dQhb0S1KL8bGch0W2n0x6l2B9nMJW7t6jwn3LqWkjN2BgL6d1vVSKf69FLZ203?t=https%3A%2F%2Finfo.conjur.net%2Fresources&si=3680029&pi=47d6aafc-d455-4a13-9ffc-a8737ef09b32>
 | Blog
<http://t.yesware.com/tt/58d6ef3f2477232db08043fa9198ec0f5a0fde51/f59ded25d9f0fae3e973353a29205cf4/c59449660ac6f9ab10a1787fd5988490/t.sidekickopen03.com/e1t/c/5/f18dQhb0S1KL8bGch0W2n0x6l2B9nMJW7t6jwn3LqWkjN2BgL6d1vVSKf69FLZ203?t=https%3A%2F%2Fblog.conjur.net%2F&si=3680029&pi=47d6aafc-d455-4a13-9ffc-a8737ef09b32>